### PR TITLE
PYIC-7286: Add app triage buffer page and improve app sub journey exi…

### PR DESCRIPTION
**Pull request by DanCorderIPV**
…t events

### What changed

New buffer page when exiting v2 app triage and clearer event names

### Why did it change

New designs from UCD

### Issue tracking

- [PYIC-7286](https://govukverify.atlassian.net/browse/PYIC-7286)

[PYIC-7286]: https://govukverify.atlassian.net/browse/PYIC-7286?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/new-p1-identity.yaml [hunk 1]
```diff
@@ -570,17 +568,15 @@ states:
   MITIGATION_01_STRATEGIC_APP_TRIAGE:
     nestedJourney: STRATEGIC_APP_TRIAGE
     exitEvents:
-      next:
+      sessionError:
```
Comment (DanCorderIPV) [inline lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/new-p1-identity.yaml:571]: I'm not sure if this is correct - do we have a definitive answer for what to do if the action to set up a v2 app session fails?

File: api-tests/features/p2-strategic-app.feature [hunk 2]
```diff
@@ -59,7 +59,11 @@ Feature: M2B Strategic App Journeys
     Then I get a 'pyi-triage-select-device' page response
     When I submit a 'smartphone' event
     Then I get a 'pyi-triage-select-smartphone' page response
-    When I submit an 'end' event
+    When I submit a 'neither' event
+    Then I get a 'pyi-triage-select-smartphone' page response
+    When I submit an 'neither' event
+    Then I get a 'pyi-triage-buffer' page response
+    When I submit an 'anotherWay' event
```
Comment (blakeyp) [inline api-tests/features/p2-strategic-app.feature]: I don't think this is quite right now this page has been updated - https://identity.build.account.gov.uk/dev/template/pyi-triage-select-smartphone/en?context=mam. The 'No' option according to the designs should now go to the DAD select smartphone page

Comment (blakeyp) [reply api-tests/features/p2-strategic-app.feature]: I'm not sure if we've captured this so we might be best raising a separate ticket for it (and just leave it how it was for now)

Comment (blakeyp) [reply api-tests/features/p2-strategic-app.feature]: Ah expanding this I can see this is exactly what this does, sorry 😄 
Can we add the page contexts now for `pyi-triage-select-smartphone`?

File: lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/new-p1-identity.yaml (hunk 1 above)
Comment (blakeyp) [reply lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/new-p1-identity.yaml:571]: Presuming this is an error response from DCMAW or something blows up in the lambda? then technical error is fine for now

File: api-tests/features/p2-strategic-app.feature (hunk 2 above)
Comment (DanCorderIPV) [reply api-tests/features/p2-strategic-app.feature]: I'll merge in main and see what needs to change

**Review by blakeyp — APPROVED** [review]

**Pull request merged**